hexrd/ui/main_window.py

```diff
@@ -165,10 +168,18 @@ def on_action_open_config_triggered(self):
             'YAML files (*.yml)')
 
         if selected_file:
+            prev_detectors = HexrdConfig().get_detector_names()
+
             HexrdConfig().load_instrument_config(selected_file)
-            self.cal_tree_view.rebuild_tree()
-            self.calibration_config_widget.update_gui_from_config()
-            self.update_all(clear_canvases=True)
+            self.update_config_gui()
+
+            new_detectors = HexrdConfig().get_detector_names()
+            if new_detectors != prev_detectors:
+                # Load the dummy images. The new config probably isn't
+                # for the current images.
+                self.load_dummy_images()
+            else:
+                self.update_all(clear_canvases=True)
```

Comment:
This section overlaps a little bit with the updates in #256, since we also want to load new dummy images when the config changes.

I'll wait until #256 is merged, then I'll re-work this, and then this PR will be ready to go.

Reply:
It is merged, and this is hopefully resolved.

Reply:
Resolved.